src/analyses/abortUnless.ml

```diff
@@ -0,0 +1,70 @@
+(** An analysis checking whether a function only returns if its only argument has a non-zero value. *)
+
+open Prelude.Ana
+open Analyses
+
+module Spec =
+struct
+  include Analyses.DefaultSpec
+
+  let name () = "abortUnless"
+  module D = BoolDomain.MustBool
+  module C = Lattice.Unit
+
+  let context _ _ = ()
+
+  (* transfer functions *)
+  let assign ctx (lval:lval) (rval:exp) : D.t =
+    false
+
+  let branch ctx (exp:exp) (tv:bool) : D.t =
+    ctx.local
```

Comment:
Any branch? Not just `arg` or `!arg`?

Reply:
Yes, branching does not actually modify the value here, so any branching is fine.